pipeline/pipelines/code_generation_pipeline.py

```diff
@@ -103,10 +116,16 @@ def do_build_flow(self, **kwargs):
                  format_tasks.PythonFormatTask('PythonFormat', inject=kwargs),
                  gapic_tasks.GapicMergeTask('GapicMerge', inject=kwargs),
                  gapic_tasks.GapicPackmanTask('GapicPackman', inject=kwargs))
+        if kwargs['publish_env'] != "none":
```

Comment:
I also prefer not changing the construction of tasks by a kwargs.

Those tasks can take the 'publish_env' argument and do nothing if not.